pkgs/by-name/au/authentik/package.nix

```diff
@@ -0,0 +1,250 @@
+{ lib
+, stdenvNoCC
+, fetchFromGitHub
+, buildNpmPackage
+, buildGoModule
+, runCommand
+, openapi-generator-cli
+, nodejs
+, python3
+, codespell
+, makeWrapper }:
+
+let
+  version = "2023.10.7";
+
+  src = fetchFromGitHub {
+    owner = "goauthentik";
+    repo = "authentik";
+    rev = "version/${version}";
+    hash = "sha256-+1IdXRt28UZ2KTa0zsmjneNUOcutP99UUwqcYyVyqTI=";
+  };
+
+  website = buildNpmPackage {
+    pname = "authentik-website";
+    inherit version src;
+    npmDepsHash = "sha256-4dgFxEvMnp+35nSQNsEchtN1qoS5X2KzEbLPvMnyR+k=";
+
+    NODE_ENV = "production";
+    NODE_OPTIONS = "--openssl-legacy-provider";
+
+    postPatch = ''
+      cd website
+    '';
+
+    installPhase = ''
+      cp -r help $out
+    '';
+
+    npmInstallFlags = [ "--include=dev" ];
+    npmBuildScript = "build-docs-only";
+  };
+
+  clientapi = stdenvNoCC.mkDerivation {
+    pname = "authentik-client-api";
+    inherit version src;
+
+    postPatch = ''
+      rm Makefile
+
+      substituteInPlace ./scripts/api-ts-config.yaml \
+        --replace-fail '/local' "$(pwd)/"
+    '';
+
+    nativeBuildInputs = [ openapi-generator-cli ];
+    buildPhase = ''
+      runHook preBuild
+      openapi-generator-cli generate -i ./schema.yml \
+      -g typescript-fetch -o $out \
+      -c ./scripts/api-ts-config.yaml \
+        --additional-properties=npmVersion=${nodejs.pkgs.npm.version} \
+        --git-repo-id authentik --git-user-id goauthentik
+      runHook postBuild
+    '';
+  };
+
+  webui = buildNpmPackage {
+    pname = "authentik-webui";
+    inherit version;
+
+    src = runCommand "authentik-webui-source" {} ''
+      mkdir -p $out/web/node_modules/@goauthentik/
+      cp -r ${src}/web $out/
+      ln -s ${src}/website $out/
+      ln -s ${clientapi} $out/web/node_modules/@goauthentik/api
+    '';
+    npmDepsHash = "sha256-5aCKlArtoEijGqeYiY3zoV0Qo7/Xt5hSXbmy2uYZpok=";
+
+    postPatch = ''
+      cd web
+    '';
+
+    installPhase = ''
+      runHook preInstall
+      mkdir $out
+      cp -r dist $out/dist
+      cp -r authentik $out/authentik
+      runHook postInstall
+    '';
+
+    NODE_ENV = "production";
+    NODE_OPTIONS = "--openssl-legacy-provider";
+
+    npmInstallFlags = [ "--include=dev" ];
```

Comment:
which is relevant here because darwin fails to build and I think we should just mark it linux only or broken on darwin.

```
Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
npm ERR! code 1
npm ERR! path /private/tmp/nix-build-authentik-webui-2023.10.7.drv-0/authentik-webui-source/web/node_modules/tree-sitter-yaml
npm ERR! command failed
npm ERR! command sh -c node-gyp rebuild
npm ERR! make: Entering directory '/private/tmp/nix-build-authentik-webui-2023.10.7.drv-0/authentik-webui-source/web/node_modules/tree-sitter-yaml/build'
npm ERR!   CXX(target) Release/obj.target/tree_sitter_yaml_binding/bindings/node/binding.o
npm ERR! make: Leaving directory '/private/tmp/nix-build-authentik-webui-2023.10.7.drv-0/authentik-webui-source/web/node_modules/tree-sitter-yaml/build'
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@10.0.1
npm ERR! gyp info using node@20.11.0 | darwin | x64
npm ERR! gyp info find Python using Python version 3.11.7 found at "/nix/store/byyb569c90slwsjr43ikhhwlhfavhiir-python3-3.11.7/bin/python3"
npm ERR! gyp WARN read config.gypi ENOENT: no such file or directory, open '/nix/store/kiprv8l36cv4c68bqpd2hcl17py4400w-nodejs-20.11.0-source/include/node/config.gypi'
npm ERR! gyp info spawn /nix/store/byyb569c90slwsjr43ikhhwlhfavhiir-python3-3.11.7/bin/python3
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args '/nix/store/zgmy5yc85d3mkz6zdmazv799c45f2gkh-nodejs-20.11.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
npm ERR! gyp info spawn args 'binding.gyp',
npm ERR! gyp info spawn args '-f',
npm ERR! gyp info spawn args 'make',
npm ERR! gyp info spawn args '-I',
npm ERR! gyp info spawn args '/private/tmp/nix-build-authentik-webui-2023.10.7.drv-0/authentik-webui-source/web/node_modules/tree-sitter-yaml/build/config.gypi',
npm ERR! gyp info spawn args '-I',
npm ERR! gyp info spawn args '/nix/store/zgmy5yc85d3mkz6zdmazv799c45f2gkh-nodejs-20.11.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
npm ERR! gyp info spawn args '-I',
npm ERR! gyp info spawn args '/nix/store/kiprv8l36cv4c68bqpd2hcl17py4400w-nodejs-20.11.0-source/common.gypi',
npm ERR! gyp info spawn args '-Dlibrary=shared_library',
npm ERR! gyp info spawn args '-Dvisibility=default',
npm ERR! gyp info spawn args '-Dnode_root_dir=/nix/store/kiprv8l36cv4c68bqpd2hcl17py4400w-nodejs-20.11.0-source',
npm ERR! gyp info spawn args '-Dnode_gyp_dir=/nix/store/zgmy5yc85d3mkz6zdmazv799c45f2gkh-nodejs-20.11.0/lib/node_modules/npm/node_modules/node-gyp',
npm ERR! gyp info spawn args '-Dnode_lib_file=/nix/store/kiprv8l36cv4c68bqpd2hcl17py4400w-nodejs-20.11.0-source/$(Configuration)/node.lib',
npm ERR! gyp info spawn args '-Dmodule_root_dir=/private/tmp/nix-build-authentik-webui-2023.10.7.drv-0/authentik-webui-source/web/node_modules/tree-sitter-yaml',
npm ERR! gyp info spawn args '-Dnode_engine=v8',
npm ERR! gyp info spawn args '--depth=.',
npm ERR! gyp info spawn args '--no-parallel',
npm ERR! gyp info spawn args '--generator-output',
npm ERR! gyp info spawn args 'build',
npm ERR! gyp info spawn args '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! gyp info spawn make
npm ERR! gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
npm ERR! In file included from ../bindings/node/binding.cc:2:
npm ERR! In file included from /nix/store/kiprv8l36cv4c68bqpd2hcl17py4400w-nodejs-20.11.0-source/src/node.h:73:
npm ERR! In file included from /nix/store/kiprv8l36cv4c68bqpd2hcl17py4400w-nodejs-20.11.0-source/deps/v8/include/v8.h:21:
npm ERR! In file included from /nix/store/cxzbqsaziwy6n2p9qz88l2vx5yns4h9n-libcxx-16.0.6-dev/include/c++/v1/memory:884:
npm ERR! In file included from /nix/store/cxzbqsaziwy6n2p9qz88l2vx5yns4h9n-libcxx-16.0.6-dev/include/c++/v1/__memory/allocate_at_least.h:13:
npm ERR! In file included from /nix/store/cxzbqsaziwy6n2p9qz88l2vx5yns4h9n-libcxx-16.0.6-dev/include/c++/v1/__memory/allocator_traits.h:14:
npm ERR! In file included from /nix/store/cxzbqsaziwy6n2p9qz88l2vx5yns4h9n-libcxx-16.0.6-dev/include/c++/v1/__memory/construct_at.h:23:
npm ERR! /nix/store/cxzbqsaziwy6n2p9qz88l2vx5yns4h9n-libcxx-16.0.6-dev/include/c++/v1/new:355:14: error: no member named 'aligned_alloc' in the global namespace
npm ERR!     return ::aligned_alloc(__alignment, __size > __rounded_size ? __size : __rounded_size);
npm ERR!            ~~^
npm ERR! 1 error generated.
npm ERR! make: *** [tree_sitter_yaml_binding.target.mk:128: Release/obj.target/tree_sitter_yaml_binding/bindings/node/binding.o] Error 1
npm ERR! gyp ERR! build error 
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack at ChildProcess.<anonymous> (/nix/store/zgmy5yc85d3mkz6zdmazv799c45f2gkh-nodejs-20.11.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:209:23)
npm ERR! gyp ERR! System Darwin 22.2.0
npm ERR! gyp ERR! command "/nix/store/zgmy5yc85d3mkz6zdmazv799c45f2gkh-nodejs-20.11.0/bin/node" "/nix/store/zgmy5yc85d3mkz6zdmazv799c45f2gkh-nodejs-20.11.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
npm ERR! gyp ERR! cwd /private/tmp/nix-build-authentik-webui-2023.10.7.drv-0/authentik-webui-source/web/node_modules/tree-sitter-yaml
npm ERR! gyp ERR! node -v v20.11.0
npm ERR! gyp ERR! node-gyp -v v10.0.1
npm ERR! gyp ERR! not ok
npm ERR! Log files were not written due to an error writing to the directory: /nix/store/z893q8w83kkz21ikd50y0k7c35ni0brz-authentik-webui-2023.10.7-npm-deps/_logs
npm ERR! You can rerun the command with `--loglevel=verbose` to see the logs in your terminal
error: builder for '/nix/store/0y5p4cmxgcmg50yy9fyannrdb8x6xaff-authentik-webui-2023.10.7.drv' failed with exit code 1;
```